### ISSUE TITLE
Not detecting missing gyro properly

### ISSUE THREAD

**Issue by jaydeetay** [sky-map-team/stardroid#153]
**Summary:** 

In v.1.9.1 some users with budget or older phones are reporting a stuck screen.   Guess is that we are not detecting the missing gyroscope and are therefore not using the 'classic' sensor path.  Might be a faulty assumption on my part about the rotation vector sensor being present even in the absence of a gyro.

**Comment by barbeau** [comment]
Rotation vector can be present when device is missing gyro - see http://stackoverflow.com/questions/7027589/which-sensors-used-for-sensor-type-rotation-vector/36454090#36454090.

**Comment by jaydeetay** [comment]
That's what I figured. The current code checks for the existence of the rotation sensor (and verified that the accelerometer and magnetic field sensor are there too because I don't trust phone manufacturers).  I'm wondering if some are claiming the existence of the sensor but just not implementing it.

**Comment by barbeau** [comment]
>I'm wondering if some are claiming the existence of the sensor but just not implementing it.

Could be, especially on older devices.  In my experience (and IMHO) CTS tests have always been on the weak side with embedded sensors.

**Comment by jaydeetay** [comment]
So far, of phones that report they have a rotation vector sensor 7% also report that they don't have a gyro.  These are the users that would have a worse experience if we used the 'classic' sensor code for gyro-less phones, and might be left in the cold completely if we in the future remove that code.

**Comment by jaydeetay** [comment]
Of course, some of those 7% will also be the users for who the rotation vector sensor is reported as present, but doesn't actually work.  Hard to know what proportion they make up.

**Comment by jaydeetay** [comment]
Worth flagging: the workaround this issue describes (some gyro-less phones falsely report having a rotation-vector sensor, so they need to be forced onto the classic sensor path) does not appear to have been ported to Sky Map v2's sensor-orientation code — no device quirk-list or fallback-forcing logic was found there. So this isn't just an old v1 bug; it looks like an unmitigated regression risk in the rewrite. Keeping open and flagging for maintainer attention. Closed #154 as a duplicate of this one (same symptom).
